### ISSUE TITLE
Feature Request: Logs Restriction Queries

### ISSUE THREAD

**Issue by christopher-dG** [DataDog/terraform-provider-datadog#1217]
Currently it seems that you cannot manage restriction queries with Terraform.

- https://docs.datadoghq.com/api/latest/logs-restriction-queries

**Comment by ohookins** [comment]
Would also really like to see this!

**Comment by zendern** [comment]
Started to look at this and quickly realized that this issue in the datadog-api-client-go needs to be completed first before this can be done. https://github.com/DataDog/datadog-api-client-go/issues/1308

**Comment by thangnc** [comment]
It seems this issue https://github.com/DataDog/datadog-api-client-go/issues/1308 to be not solved and released yet

**Comment by djtecha** [comment]
Any updates on this? Would be useful to have.

**Comment by nemobis** [comment]
Docs still state

> Note: This endpoint is in public beta. If you have any feedback, contact [Datadog support](https://web.archive.org/web/20220123084410/https://docs.datadoghq.com/help/).

**Comment by jbaker10** [comment]
Reviving this thread to see where this stands.

**Comment by milpog** [comment]
We would be also interested in having that feature implemented.

**Comment by diegoboff** [comment]
Would love to see this being implemented too

**Comment by jyrkihuhta** [comment]
Any progress with this? This feature is quite important!

**Comment by fd-mikeli** [comment]
would be super helpful

**Comment by wsellars-bw** [comment]
Still waiting on the go client it looks like

**Comment by nedkoh** [comment]
Would really want to see this one completed. We are using Terraform in production and this gap really limits our capabilities

**Comment by danilogomesrosa** [comment]
Do we have any progress?

**Comment by AtomicBackspace** [comment]
I assume this was completed with the latest release? Or am I misunderstanding this: https://github.com/DataDog/terraform-provider-datadog/pull/3352 ?

**Comment by hi-artem** [comment]
Yes this should be resolved now https://registry.terraform.io/providers/datadog/datadog/latest/docs/resources/logs_restriction_query 🙌

**Comment by teamRocket-Miaouss** [comment]
resource is ok !!! but i need "data" now :/
it's in progress ?
